addons/mail/static/src/core/common/thread_service.js

```diff
@@ -615,13 +615,12 @@ export class ThreadService {
             typeof thread.id === "string"
                 ? `mail.box_${thread.id}`
                 : `discuss.channel_${thread.id}`;
-        this.store.discuss.activeTab = !this.ui.isSmall
-            ? "all"
-            : thread.model === "mail.box"
-            ? "mailbox"
-            : ["chat", "group"].includes(thread.type)
-            ? "chat"
-            : "channel";
+        this.store.discuss.activeTab =
+            !this.ui.isSmall || thread.model === "mail.box"
+                ? "main"
+                : ["chat", "group"].includes(thread.type)
+                ? "chat"
+                : "channel";
```

Comment:
Can we please refactor this with something readable?

Reply:
This is a fix PR, not code cleaning. There's a few code cleaning done when I feel it helps the fix, this should be more of a bonus than a requirement for now

Reply:
But you have diff on it, it's a good opportunity

Reply:
If the motivation is you don't like it (which is not necessarily wrong, I get it too that's it's not ugly and there might be better LOCs to do that) but there's no solution proposal or more genuine other concerns, then this is not worth improving in this PR for now